pkg/apis/pipeline/v1alpha1/pipelinerun_types_test.go

```diff
@@ -0,0 +1,64 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/knative/pkg/apis"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestPipelineRunStatusConditions(t *testing.T) {
```

Comment:
@shashwathi could you explain a bit more in the commit message and pull description (and maybe a comment here too) a bit more about why this test was added?

im guessing this has something to do with supporting duck typing, but i'm not ramped up on it enough to understand why we need this test (esp. when we only have one condition at the moment). is this actually a test of the `pkg` Conditions, or is this required to make sure we are using them properly?

Reply:
`TestPipelineRunStatusConditions` tests that `SetCondition` function on pipeline status can be used to set any ducktype condition and similarly `GetCondition` function can be used to fetch any ducktype condition. I added 2 conditions to demonstrate that we are not limiting to just setting one condition even though pipeline run uses only one condition today.

> is this required to make sure we are using them properly?

Not required. I think the test demonstrates we are using them correctly. Required test to verify pipelinerun status complies with ducktype interface.(which I have not done in this PR)

I agree that this is not part of this PR scope. I was adding tests to pipeline run so took the opportunity to add coverage for the struct. I will update the PR description.